lib/shopify_transporter/shopify/customer.rb

```diff
@@ -57,12 +57,28 @@ def to_csv
       ).freeze
 
       def top_level_row_values
-        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES)).values
+        default_address.merge(top_level_attributes).values_at(*Customer.columns)
+      end
+
+      def top_level_attributes
+        base_hash
+          .merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES))
+          .delete_if { |_, v| v.nil? }
```

Comment:
Is there a specific reason to add `.delete_if { |_, v| v.nil? }`? If it doesn't add an obvious benefit to the csv output, let's remove it.

Otherwise, I suggest adding a test case for this.

Reply:
Yep there's a reason for that. 

In function `top_level_row_values`, I'm merging `top_level_attributes` into `default_address`. Before the merge happens, this `delete_if` makes sure that all `nil` values of `top_level_attributes` are removed. Otherwise, the `nil` address fields from `top_level_attributes` will overwrite existing values in `default_address`. 

The reason why I'm merging `top_level_attributes` into `default_address`, instead of the other direction around, is that I want to make sure fields like `name` and `email`in `top_level_attributes` take precedence over fields in `default_address`.

I believe `delete_if` is already covered by existing test cases as suggested by coverage report.